feat(settings): add settings page with display name, profile image, and Chiko

NathanielDSmith opened the pull request:
✅ Resolves #[Issue number]
- [ X ] PR title follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specifications
- [ X ] PR assignee has been selected
- [ X ] PR label has been selected
- [ X ] @NabbeunNabi has been selected for preliminary review

## 🔧 What changed
- Created basic settings page layout
- Included input for display name and profile image
- Added Chiko therapy dog SVG at bottom of the page
- Styled components using Tailwind

## 🧪 Testing instructions
This component is not yet integrated into the app. To view it, navigate manually to the /settings route or temporarily wire it up in the router for preview. Styling and structure can be confirmed visually.

## 📸 Screenshots
-   ### Before
File didn't exist prior.
-   ### After
<img width="1898" height="877" alt="image" src="https://github.com/user-attachments/assets/75c093bb-9407-4cc5-a723-b76e1afde321" />

NabbeunNabi commented on pages/settings.vue:
```diff
@@ -0,0 +1,55 @@
+    <div class="min-h-screen bg-primary-bg pt-20 px-4">
+        <div class="max-w-2xl mx-auto">
+            <div class="mb-8">
+                <h1 class="text-3xl font-bold text-primary">Settings</h1>
```
We want these to use translatable strings.

Look at other components or ask in slack #tech-help for some guidance

NabbeunNabi commented on pages/settings.vue:
```diff
@@ -0,0 +1,55 @@
+        <div class="max-w-2xl mx-auto">
+            <div class="mb-8">
+                <h1 class="text-3xl font-bold text-primary">Settings</h1>
+                <p class="text-gray-600">Manage your account settings</p>
```
We don't want to use hardcoded colors. We want to use our theme colors. You can look into other components as well or look.

Things like `text-primary`

Also I am not sure if p is the correct tag semantically as it isn't a paragraph 🤔

NabbeunNabi commented on pages/settings.vue:
```diff
@@ -0,0 +1,55 @@
+                <p class="text-gray-600">Manage your account settings</p>
+            </div>
+
+            <div class="bg-white rounded-lg shadow p-6 mb-8">
```
♻️ means I referred to it before. But we should not have hardcoded colors. Only theme colors

NabbeunNabi commented on pages/settings.vue:
```diff
@@ -0,0 +1,55 @@
+
+            <div class="bg-white rounded-lg shadow p-6 mb-8">
+                <div class="mb-6">
+                    <h2 class="text-xl font-semibold text-primary mb-3">Display Name</h2>
```
♻️ translatable strings

NabbeunNabi commented on pages/settings.vue:
```diff
@@ -0,0 +1,55 @@
+                    <input
+                        type="text"
+                        value="John Doe"
+                        placeholder="Enter your display name"
```
♻️ translatable strings. Might need to use :placeholder to be able to take a variable

NabbeunNabi commented on pages/settings.vue:
```diff
@@ -0,0 +1,55 @@
+                    <h2 class="text-xl font-semibold text-primary mb-3">Display Name</h2>
+                    <input
+                        type="text"
+                        value="John Doe"
```
This doesn't need to be functional yet as in autofilled. But let's try to add `v-model` and a `ref` to look at what these are in vue and try to use them

NabbeunNabi commented on pages/settings.vue:
```diff
@@ -0,0 +1,55 @@
+                        type="text"
+                        value="John Doe"
+                        placeholder="Enter your display name"
+                        class="w-full px-4 py-3 border border-gray-300 rounded-lg"
```
♻️ use theme colors 🙇

NabbeunNabi commented on pages/settings.vue:
```diff
@@ -0,0 +1,55 @@
+                </div>
+
+                <div class="mb-6">
+                    <h2 class="text-xl font-semibold text-primary mb-3">Profile Image</h2>
```
♻️ translatable strings

NabbeunNabi commented on pages/settings.vue:
```diff
@@ -0,0 +1,55 @@
+                        <img
+                            src="~/assets/icons/profile-icon.svg"
+                            alt="profile icon"
+                            class="w-20 h-20 rounded-full border-2 border-gray-200"
```
♻️ theme colors. I think in terms of a picture let's use Chiko here as well for fun 😄

NabbeunNabi commented on pages/settings.vue:
```diff
@@ -0,0 +1,55 @@
+                    <h2 class="text-xl font-semibold text-primary mb-3">Profile Image</h2>
+                    <div class="flex items-center gap-6">
+                        <img
+                            src="~/assets/icons/profile-icon.svg"
```
SVGs do not need to be the source of an img tag in vue. You can find other places they are imported and used. Usually starts with `SVGProfileImage` or something similar

NabbeunNabi commented on pages/settings.vue:
```diff
@@ -0,0 +1,55 @@
+                            alt="profile icon"
+                            class="w-20 h-20 rounded-full border-2 border-gray-200"
+                        />
+                        <div class="flex-1">
+                            <input
+                                type="text"
+                                placeholder="Enter image URL"
+                                class="w-full px-4 py-3 border border-gray-300 rounded-lg"
+                            />
+                            <p class="text-sm text-gray-500 mt-2">Enter a URL to your profile image</p>
+                        </div>
+                    </div>
+                </div>
```
You can remove this for now and just display the image as it will be an upload file button when given functionality and this will be deleted

NabbeunNabi commented on pages/settings.vue:
```diff
@@ -0,0 +1,55 @@
+                </div>
+
+                <div class="flex justify-end">
+                    <button class="px-6 py-3 bg-primary text-white rounded-lg font-medium">
```
♻️ theme colors

NabbeunNabi commented on pages/settings.vue:
```diff
@@ -0,0 +1,55 @@
+
+                <div class="flex justify-end">
+                    <button class="px-6 py-3 bg-primary text-white rounded-lg font-medium">
+                        Update Settings
```
♻️ translatable strings

NabbeunNabi commented on pages/settings.vue:
```diff
@@ -0,0 +1,55 @@
+
+            <div class="text-center">
+                <img
+                    src="~/assets/icons/characters-chico-therapy-dog-primary.svg"
```
Chiko is the correct spelling as you listed below. So let's change the file name to be the same as the correct spelling by and fix my typo from the file name 😂

NabbeunNabi reviewed (changes requested):
Great start @NathanielDSmith let's try to add a script tag and look at some other components for a little inspiration for the suggested changes

NathanielDSmith commented:
Hey! I believe I’ve fixed the filename to use the correct spelling (Chiko), and updated the reference in the code too.

This is actually the first time I’ve received direct feedback on my code, so I really appreciate the opportunity. I’m also learning how to use GitHub properly as I go, since this is my first time dealing with a pull request like this. I know my code might look rough in places, but I’m doing my best to improve and study Vue more seriously now. Thanks again for your patience and help.

NabbeunNabi commented:
@NathanielDSmith I am glad that we are able to give you the direct feedback. I would be surprised if you came in and got something merged without some feedback. It is a common practice and the way you handled it is perfect. I work on stuff at my job that sometimes takes many many back and forths before it is ready. It is more rare to not get direct feedback.

If you have fixed everything don't forget to ping me by clicking this button in the photo I am pasting. It will send me an email. Or post it in slack in #code-reviews a channel you can join

<img width="327" height="67" alt="image" src="https://github.com/user-attachments/assets/4899f516-c52a-4b90-afc1-f46345ed6a70" />

It appears you have some linting errors. You will want to run `yarn lint:fix` locally to fix them and then push the changes

NabbeunNabi commented:
<img width="595" height="187" alt="image" src="https://github.com/user-attachments/assets/86d7c28a-ee1e-471b-812f-7fb4ac9c83b7" />
Also for this. If you remove the space around the `X` you will see a checkbox like the one I will show below with the markdown

Before:
- [ X ]

After
- [x]

NabbeunNabi commented on pages/settings.vue:
```diff
@@ -0,0 +1,58 @@
+    <div class="min-h-screen bg-primary-bg pt-20 px-4">
+        <div class="max-w-2xl mx-auto">
+            <div class="mb-8">
+                <h1 class="text-3xl font-bold text-primary">{{ t('submitPage.heading') }}</h1>
+                <span class="text-primary-text-muted">{{ t('submitPage.subheading') }}</span>
```
Nice update. But these are settings page. So you will want to add the translations to the `en.json` file with its own key value. And then you can add all the translations once you do this for all the translations in this file with `yarn lint:locales`

NabbeunNabi commented on pages/settings.vue:
```diff
@@ -0,0 +1,58 @@
+                    <input
+                        type="text"
+                        v-model="displayName"
+                        :placeholder="t('submitPage.firstName')"
```
I actually do not think a placeholder is necessary here as this will be autofilled with their name that they get when signing up and will not have access to it. Sorry to contradict myself by you can remove the placeholder

NabbeunNabi commented on pages/settings.vue:
```diff
@@ -0,0 +1,58 @@
+                </div>
+
+                <div class="mb-6">
+                    <h2 class="text-xl font-semibold text-primary mb-3">{{ t('submitPage.healthcareProfessionalName') }}</h2>
```
♻️

NabbeunNabi commented on pages/settings.vue:
```diff
@@ -0,0 +1,58 @@
+                <div class="mb-6">
+                    <h2 class="text-xl font-semibold text-primary mb-3">{{ t('submitPage.healthcareProfessionalName') }}</h2>
+                    <div class="flex items-center gap-6">
+                        <SVGProfileIcon
+                            role="img"
+                            alt="profile icon"
+                            title="profile icon"
+                            class="w-20 h-20 rounded-full border-2 border-primary-text-muted stroke-primary"
+                        />
```
use Chiko for fun 😸

NabbeunNabi commented on pages/settings.vue:
```diff
@@ -0,0 +1,58 @@
+
+                <div class="flex justify-end">
+                    <button class="px-6 py-3 bg-primary text-primary-inverted rounded-lg font-medium">
+                        {{ t('submitPage.submitButton') }}
```
♻️

NabbeunNabi commented on pages/settings.vue:
```diff
@@ -0,0 +1,58 @@
+
+            <div class="text-center">
+                <img
+                    src="~/assets/icons/characters-chiko-therapy-dog-primary.svg"
```
Same as before. This is an SVG so can be imported the same way.

NabbeunNabi reviewed (changes requested):
Big improvement. A big more to go. Also i am curious. What does this look like on mobile 🤔

NathanielDSmith commented:
I believe I’ve fixed everything that was mentioned. Thanks again for the feedback so far.

If you notice anything else that needs tweaking, feel free to let me know.

Just a heads-up though: this has been a pretty tricky and time-consuming process for me, especially with my limited experience in Vue and the project structure. I'm still learning as I go, so I really appreciate your patience while I try to get things right. 🙏

Mobile view
<img width="267" height="582" alt="settings page mobile" src="https://github.com/user-attachments/assets/239f4537-4f8c-4727-a31a-9508f51acc90" />

NabbeunNabi commented:
> I believe I’ve fixed everything that was mentioned. Thanks again for the feedback so far.
> 
> 
> 
> If you notice anything else that needs tweaking, feel free to let me know.
> 
> 
> 
> Just a heads-up though: this has been a pretty tricky and time-consuming process for me, especially with my limited experience in Vue and the project structure. I'm still learning as I go, so I really appreciate your patience while I try to get things right. 🙏
> 
> 
> 
> Mobile view
> 
> <img width="267" height="582" alt="settings page mobile" src="https://github.com/user-attachments/assets/239f4537-4f8c-4727-a31a-9508f51acc90" />
> 
> 

@NathanielDSmith do not worry about it taking you a lot of time, or being new to everything.

We are here to help and be supportive through the learning process.

If you ever feel stuck please reach out to us in slack in the frontend channel or tech-help.

We take all levels and just appreciate the interest and work.

NabbeunNabi commented:
@NathanielDSmith Just a UI nit. Would you mind flipping the picture and display name. Since if there are other parts of the form that are inputs they would go under and it would be good to be together?

NathanielDSmith commented:
Reordered the display name and profile image as requested.
Attached are updated screenshots for both desktop and mobile views. Let me know if anything else needs tweaking.
<img width="1902" height="920" alt="settings page desktop" src="https://github.com/user-attachments/assets/57502436-9777-47ce-ad14-1d24502cf5cf" />
<img width="267" height="580" alt="settings page mobile" src="https://github.com/user-attachments/assets/91a5e7d4-c22c-4272-bcbe-84912a399ecf" />

NabbeunNabi commented:
This was a fork. A new PR was made here: https://github.com/ourjapanlife/findadoc-web/pull/1393 

This is closed but referenced there for context
